File: 1200/1537C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;
        vector<int> a(n);

        for (int i = 0; i < n; i++)
            cin >> a[i];

        sort(a.begin(), a.end());

        int idx1 = -1;
        int diff = INT_MAX;

        for (int i = 0; i < n - 1; i++) 
        {
            if (diff > a[i + 1] - a[i]) 
            {
                idx1 = i;
                diff = a[i + 1] - a[i];
            }
        }

        cout << a[idx1] << " ";
        for (int i = idx1 + 2; i < n; i++)
            cout << a[i] << " ";
    
        for (int i = 0; i < idx1; i++) 
            cout << a[i] << " ";
        
        cout << a[idx1 + 1] << "\n";
    }
    return 0;
}

File: 1200/1537C/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int a[] = new int[n];
            
            for(int i=0;i<n;i++)
                a[i] = sc.nextInt();
            
            Arrays.sort(a);
            
            int idx1 = -1;
            int diff = 1000000000;
            
            for(int i=0;i<n-1;i++)
            {
                if(diff > a[i+1]-a[i])
                {
                    idx1 = i;
                    diff = a[i+1]-a[i];
                }
            }
            
            System.out.print(a[idx1]+" ");
            for(int i=idx1+2;i<n;i++)
                System.out.print(a[i]+" ");
            for(int i=0;i<idx1;i++)
                System.out.print(a[i]+" ");
            System.out.println(a[idx1+1]);
        }
    }
}